lib/beaker/hypervisor/docker.rb

```diff
@@ -65,6 +65,19 @@ def provision
         host['docker_container'] = container
         host['docker_image'] = image
       end
+
+      hack_docker_etc_hosts @hosts, @options
+
+    end
+
+    def hack_docker_etc_hosts hosts, opts
+      etc_hosts = "127.0.0.1\tlocalhost localhost.localdomain\n"
+      hosts.each do |host|
+        etc_hosts += "#{host['docker_container'].json["NetworkSettings"]["IPAddress"]}\t#{host.name}\n"
```

Comment:
hmm.. yes, but the `hack_etc_hosts` method isn't present in the `provision` method, is it? :)
or have you added it? When I add `hack_etc_hosts` in the `provision` method in https://github.com/puppetlabs/beaker/blob/master/lib/beaker/hypervisor/docker.rb#L26, it creates a hostfile with `0.0.0.0` listed for all hosts